src/Arcus.WebApi.Correlation/CorrelationInfo.cs

```diff
@@ -0,0 +1,33 @@
+﻿using GuardNet;
+using Microsoft.AspNetCore.Http;
+
+namespace Arcus.WebApi.Correlation
+{
+    /// <summary>
+    /// Represents the correlation information on the current HTTP request, accessible throughout the application.
+    /// </summary>
+    public class CorrelationInfo
+    {
+        private readonly CorrelationFeature _correlationFeature;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CorrelationInfo"/> class.
+        /// </summary>
+        public CorrelationInfo(IHttpContextAccessor contextAccessor)
```

Comment:
Since we'll use this class for messaging as well, is it possible that we populate this information via the ctor instead of `IHttpContextAccessor`? Or can we introduce a `HttpCorrelationInfo` that derives from `CorrelationInfo`?

Reply:
I renamed the `CorrelationFeature` public and renamed it to `CorrelationInfo` + I renamed the `CorrelationInfo` to `HttpCorrelationInfo`. That way, we can reuse the `CorrelationInfo` class in messaging.

Reply:
Perfect, thanks!